nautobot/dcim/models/device_components.py

```diff
@@ -515,6 +515,98 @@ class BaseInterface(RelationshipModel):
     class Meta:
         abstract = True
 
+    def clean(self):
+        super().clean()
+
+        # Virtual interfaces cannot be connected
+        if getattr(self, "type", None) in NONCONNECTABLE_IFACE_TYPES and (
+            self.cable or getattr(self, "circuit_termination", False)
+        ):
+            raise ValidationError(
+                {
+                    "type": "Virtual and wireless interfaces cannot be connected to another interface or circuit. "
+                    "Disconnect the interface or choose a suitable type."
+                }
+            )
+
+        # Parent validation
+        if self.parent_interface is not None:
+
+            # An interface cannot be its own parent
+            if self.parent_interface_id == self.pk:
+                raise ValidationError({"parent_interface": "An interface cannot be its own parent."})
+            # A physical interface cannot have a parent interface
+            if hasattr(self, "type") and self.type != InterfaceTypeChoices.TYPE_VIRTUAL:
+                raise ValidationError(
+                    {"parent_interface": "Only virtual interfaces may be assigned to a parent interface."}
+                )
+
+            # A virtual interface cannot be a parent interface
+            if getattr(self.parent_interface, "type", None) == InterfaceTypeChoices.TYPE_VIRTUAL:
+                raise ValidationError(
+                    {"parent_interface": "Virtual interfaces may not be parents of other interfaces."}
+                )
+
+            # An interface's parent must belong to the same device or virtual chassis
+            if self.parent_interface.parent != self.parent:
+                if self.parent.virtual_chassis is None:
+                    raise ValidationError(
+                        {
+                            "parent_interface": f"The selected parent interface ({self.parent_interface}) belongs to a different device "
+                            f"({self.parent_interface.device})."
+                        }
+                    )
+                elif self.parent_interface.device.virtual_chassis != self.parent_interface.virtual_chassis:
```

Comment:
Is this correct? I would have thought it should be:

`elif self.parent_interface.parent.virtual_chassis != self.parent.virtual_chassis`